service/resource/deployment/deployment.go

```diff
@@ -50,7 +50,12 @@ func (r Resource) newMainDeployment(obj providerv1alpha1.AzureConfig) (deploymen
 		workerNodes = append(workerNodes, n)
 	}
 
-	masterCloudConfig, err := r.cloudConfig.NewMasterCloudConfig(obj)
+	cluster, err := r.randomkeysSearcher.SearchCluster(key.ClusterID(obj))
+	if err != nil {
+		return deployment{}, microerror.Mask(err)
+	}
```

Comment:
Wouldn't it be nice to encapsulate that knowledge in `cloudconfig` package w/o interface changes?

Reply:
yeah make sense, so we would not even touch the signature of `NewMasterCloudConfig`

Reply:
Yes, that would be the goal.